lib/hexpm/accounts/users.ex

```diff
@@ -383,6 +385,18 @@ defmodule Hexpm.Accounts.Users do
     end
   end
 
+  def tfa_recover(%User{} = user, code_str) do
+    case Recovery.verify(user.tfa.recovery_codes, code_str) do
+      {:ok, %RecoveryCode{} = code} ->
```

Comment:
will it match ever if `Repo.update!/2` doesn't return struct but schema directly or raises an error

Reply:
cc @supersimple @starbelly